### PR TITLE
fix: Replace RequestError with BadRequestError

### LINKED ISSUE #3169: Replace elasticsearch RequestError with BadRequestError
## Description

`api/controllers/search_controller` imports an exception (`RequestError`) that does not exist:

https://github.com/WordPress/openverse/blob/3bb5bff6d0cf9cd381405c1e0392bb06a1f51783/api/api/controllers/search_controller.py#L12

Replace `RequestError` with `BadRequestError`:
~~`from elasticsearch.exceptions import NotFoundError, RequestError`~~
`from elasticsearch.exceptions import NotFoundError, BadRequestError`

## Additional context

[2 years ago](https://github.com/elastic/elasticsearch-py/commit/6462574ba44a87075a37a9e810ed8675f9511e76), the code update in Elasticsearch was updated and `RequestError` was replaced with `BadRequestError`.

### PR DESCRIPTION
**Pull request by CodeAnk2829**
Fixes #3169 by @CodeAnk2829

## Description
`api/controllers/search_controller` imports an exception (`RequestError`) that does not exist causes an unexpected error. This PR fixes the issue by replacing `RequestError` with `BadRequestError`.

## Checklist

- [x] My pull request has a descriptive title (not a vague title like`Update index.md`).
- [x] My pull request targets the _default_ branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.
- [ ] I ran the DAG documentation generator (if applicable).

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: api/api/controllers/search_controller.py
Comment (dhruvkb) [inline api/api/controllers/search_controller.py:12]: I believe the lint error we're seeing is about the order of the imports. Imports are sorted alphabetically so placing BadRequestError first should fix it.

**Review by dhruvkb — APPROVED** [review]
The changes look good to me except the lint error. Approving to unblock merge once this is fixed (failing CI is blocking the merge already).

File: api/api/controllers/search_controller.py
Comment (CodeAnk2829) [reply api/api/controllers/search_controller.py:12]: Understood @dhruvkb, will change in a while.

**Review by obulat — APPROVED** [review]
I ran `just lint` and committed the changes (the imports were sorted alphabetically).

Thank you for your contribution, and also for finding and updating the exceptions in documentation!

**Comment by CodeAnk2829** [comment]
Thank you so much for you appreciation @obulat :)

**Pull request merged**
